### DISCUSSION TITLE
Is there a State which can be used for checking whether the ref url is the currently active url?

### DISCUSSION THREAD

**Discussion by schlichtanders** [tailwindlabs/tailwindcss#5773] [Help] [answered]
I would like to have a navbar where the currently active page is also highlighted in the navbar.

I couldn't find anything useful by googling, but [something from ruby](https://gorails.com/forum/what-is-the-best-way-to-add-a-dynamic-active-class-to-a-link-with-tailwind-css) where they recommend to compare the href path to the currently active path.

That sounds simple enough to be checked by tailwind so I am curious whether this can be done, or if not, whether this might be done within future tailwind.

**Comment by simonswiss** [comment, marked as answer]
Hey!

This is typically something you'd do with a templating language/framework helper (like in the Ruby `.erb` template example you shared), or with some JavaScript on the client-side.

It's possible to do only with CSS, but you'd need to change classes on say the `body` tag for each page to then style a given link differently based on the page:

And even there, I'd say this should not the responsibility of Tailwind CSS to determine that, Tailwind's role would be to provide the appropriate utility classes to style your active link.

But if you wanted to do it with Tailwind, and are able to change the class on the `body` (or parent element from navigation), you could do:

```html
<!-- on every page... -->
<nav>
  <a class="home" href="/">Home</a>
  <a class="about" href="/about">About</a>
  <a class="contact" href="/contact">Contact</a>
</nav>
```

```html
<!-- Homepage -->
<body class="home">...</body>

<!-- About page -->
<body class="about">...</body>

<!-- Contact page -->
<body class="contact">...</body>
```

Then, in your CSS:

```css
body.home nav a.home,
body.about nav a.about,
body.contact nav a.contact {
  @apply bg-red-500 font-bold;
}
```

That said, I'd still recommend doing it at the templating level instead, as this will add specificity to your CSS, or accidentally style something else (unless you give a custom class to your navigation). But the possibility is there, if you want to.
